### ISSUE TITLE
Nullness checker incorrectly thinks BlockingQueue.poll( long, TimeUnit ) returns @NonNull

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#374]
```
This method can return null. It should be treated as @Nullable but the Nullness 
checker thinks it's @NonNull

Thus I get the spurious error:

[ERROR] 
/home/david/apache/Blackhawk/javatestframework/util/src/main/java/com/trutest/ut
il/ProfiledBlockingQueue.java:[58,21] [override.return.invalid] @Nullable E 
extends @Initialized @Nullable Object poll(@Initialized @NonNull 
ProfiledBlockingQueue<E extends @Initialized @NonNull Object> this, 
@Initialized @NonNull long p0, @Initialized @NonNull TimeUnit p1) throws 
@Initialized @NonNull InterruptedException in 
com.trutest.util.ProfiledBlockingQueue cannot override E extends @Initialized 
@NonNull Object poll(@Initialized @NonNull BlockingQueue<E extends @Initialized 
@NonNull Object> this, @Initialized @NonNull long p0, @Initialized @NonNull 
TimeUnit p1) throws @Initialized @NonNull InterruptedException in 
java.util.concurrent.BlockingQueue; attempting to use an incompatible return 
type


using checker 1.8.6 on Java 8 on Linux.

Please provide any additional information below.

```

Original issue reported on code.google.com by `DavidHod...@gmail.com` on 16 Oct 2014 at 12:30
